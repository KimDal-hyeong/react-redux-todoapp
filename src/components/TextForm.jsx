import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  className: PropTypes.string,
  onChange: PropTypes.func,
  onSubmit: PropTypes.func,
  placeholder: PropTypes.string,
};

const defaultProps = {
  className: '',
  onChange() {},
  onSubmit() {},
  placeholder: '',
};

function TextForm(props) {
  const {
    className,
    onChange,
    onSubmit,
    placeholder,
  } = props;

  function handleKeypress(e) {
    if(e.key === 'Enter') {
      onSubmit(e.target.value);
      e.target.value = '';
    }
  }

  return (
    <input
      className={className}
      placeholder={placeholder}
      type="text"
      onChange={e => {onChange(e.target.value)}}
      onKeyPress={handleKeypress}
      {...props}
    />
  );
}

TextForm.propTypes = propTypes;
TextForm.defaultProps = defaultProps;

export default TextForm;