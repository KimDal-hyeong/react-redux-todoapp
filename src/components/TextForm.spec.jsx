import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import TextForm from './TextForm';

describe('<TextForm />', () => {
  const mockClassName = 'mock-class-name';
  const mockPlaceholder = 'mock-title';
  const mockValue = 'mock-value';
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<TextForm
      className={mockClassName}
      placeholder={mockPlaceholder}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClose when press Enter', () => {
    const onSubmit = jest.fn();
    const wrapper = shallow(<TextForm
      placeholder={mockPlaceholder}
      onSubmit={onSubmit}
    />);
    const input = wrapper.find('input').at(0);

    input.simulate('keypress', {
      key: 'Enter',
      target: {
        value: mockValue
      }
    });

    expect(onSubmit.mock.calls.length).toBe(1);
  });
});