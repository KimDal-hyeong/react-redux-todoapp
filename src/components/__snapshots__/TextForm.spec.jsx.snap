// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextForm /> should match to snapshot when render default 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextForm
    className="mock-class-name"
    onChange={[Function]}
    onSubmit={[Function]}
    placeholder="mock-title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "mock-class-name",
      "onChange": [Function],
      "onKeyPress": [Function],
      "onSubmit": [Function],
      "placeholder": "mock-title",
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "mock-class-name",
        "onChange": [Function],
        "onKeyPress": [Function],
        "onSubmit": [Function],
        "placeholder": "mock-title",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
